runtime-modules/bounty/src/lib.rs

```diff
@@ -1118,127 +1377,210 @@ decl_module! {
+            Self::deposit_event(RawEvent::OracleJudgmentSubmitted(bounty_id, bounty.creation_params.oracle, judgment));
         }
 
-        /// Withdraw work entrant funds.
-        /// Both legitimate participants and winners get their stake unlocked. Winners also get a
-        /// bounty reward.
+        ///Unlocks the stake related to a work entry
+        ///After the oracle makes the judgment or the council terminates the bounty by calling terminate_bounty(...),
+        ///each worker whose entry has not been judged, can unlock the totality of their stake.
         /// # <weight>
         ///
         /// ## weight
         /// `O (1)`
         /// - db:
         ///    - `O(1)` doesn't depend on the state or parameters
         /// # </weight>
-        #[weight = WeightInfoBounty::<T>::withdraw_work_entrant_funds()]
-        pub fn withdraw_work_entrant_funds(
+        #[weight = WeightInfoBounty::<T>::unlock_work_entrant_stake()]
+        pub fn unlock_work_entrant_stake(
             origin,
             member_id: MemberId<T>,
             bounty_id: T::BountyId,
             entry_id: T::EntryId,
         ) {
-            let controller_account_id =
-                T::Membership::ensure_member_controller_account_origin(origin, member_id)?;
+            T::Membership::ensure_member_controller_account_origin(origin, member_id)?;
 
             let bounty = Self::ensure_bounty_exists(&bounty_id)?;
 
             let current_bounty_stage = Self::get_bounty_stage(&bounty);
 
-            // Ensure withdrawal for successful or failed bounty.
             ensure!(
-                current_bounty_stage == BountyStage::FailedBountyWithdrawal ||
-                current_bounty_stage == BountyStage::SuccessfulBountyWithdrawal,
+                matches!(current_bounty_stage,
+                    BountyStage::FailedBountyWithdrawal |
+                    BountyStage::SuccessfulBountyWithdrawal ),
                 Self::unexpected_bounty_stage_error(current_bounty_stage)
             );
 
             let entry = Self::ensure_work_entry_exists(&entry_id)?;
 
+            Self::ensure_work_entry_ownership(&entry, &member_id)?;
+
             //
             // == MUTATION SAFE ==
             //
 
-            // Claim the winner reward.
-            if let Some(OracleWorkEntryJudgment::Winner { reward }) = entry.oracle_judgment_result {
-                Self::transfer_funds_from_bounty_account(
-                    &controller_account_id,
+            T::StakingHandler::unlock(&entry.staking_account_id);
+            Self::deposit_event(
+                RawEvent::WorkEntrantStakeUnlocked(
                     bounty_id,
-                    reward
-                )?;
+                    entry_id,
+                    entry.staking_account_id)
+            );
+            Self::remove_work_entry(&bounty_id, &entry_id);
+
+            if Self::withdrawal_completed(&current_bounty_stage, &bounty_id) {
```

Comment:
Why can we ignore destination of cherry and oracle funds in all cases?

Reply:
Could you explain this point better, please